lib/bandit/http2/frame/settings.ex

```diff
@@ -33,10 +33,10 @@ defmodule Bandit.HTTP2.Frame.Settings do
         {:cont, {:ok, %{acc | header_table_size: value}}}
 
       {:ok, {0x02, 0x01}}, {:ok, acc} ->
-        {:cont, {:ok, %{acc | enable_push: true}}}
+        {:cont, {:ok, acc}}
```

Comment:
Can we remove these cases all together (and ignore invalid `enable_push` values)?
`h2spec` seems fine with it.

Otherwise group the valid cases to `{:ok, {0x02, val}}, {:ok, acc} when val in [0x00, 0x01] ->`?

Reply:
I'd like to keep the handling, even if h2spec doesn't complain. No sense in walking back strict conformance. 

Updated to use a when clause, as suggested

Reply:
Or maybe use `{:ok, {0x02, value}}, {:ok, _acc} when value not in [0x00, 0x01]` on the error case and drop the noop case?